lib/std/zig/CrossTarget.zig

```diff
@@ -231,6 +231,32 @@ pub const ParseOptions = struct {
     };
 };
 
+/// Parses versions with omitted patch components, such as "1.0",
+/// which SemanticVersion.parse is not capable of.
+fn parseVersion(string: []const u8) !SemanticVersion {
+    const parseVersionComponent = struct {
+        fn parseVersionComponent(component: []const u8) !usize {
+            return std.fmt.parseUnsigned(usize, component, 10) catch |err| {
+                switch (err) {
+                    error.InvalidCharacter => return error.InvalidVersion,
+                    error.Overflow => return error.Overflow,
+                }
+            };
+        }
+    }.parseVersionComponent;
+    var version_components = mem.split(u8, string, ".");
+    const major = version_components.first();
+    if (major.len == 0) return error.InvalidVersion;
```

Comment:
Why is this only checked for the first component?

Reply:
I think we can actually remove this `if`. `parseUnsigned` will catch the case of an empty string for all three versions components.

I will actually add tests for all this to ensure that. You're right that this could use some tests.